source/update.rst

```diff
@@ -1,26 +1,101 @@
-Update Percona XtraDB Cluster Operator
+.. _operator-update:
+
+Upgrade Percona XtraDB Cluster
 ======================================
 
-Starting from the version 1.1.0 the Percona Kubernetes Operator for Percona XtraDB
-Cluster allows upgrades to newer versions. This upgrade can be done either in
-semi-automatic or in manual mode. 
+Starting from version 1.1.0, Percona Kubernetes Operator for Percona
+XtraDB Cluster allows upgrades to newer versions. The upgrade can be done in
+three ways: automatic (also called *Smart Update*), semi-automatic, or manual.
+
+.. note:: Smart Update mode is the recommended way for a production cluster.
+
+.. contents:: :local:
+
+.. _operator-update-smartupdates:
+
+Automatic upgrade
+-----------------
+
+Starting from version 1.5.0, the Operator can do fully automatic upgrades to
+newer versions within the method named *Smart Updates*.
+
+To have this upgrade method enabled, make sure that the ``updateStrategy`` key
+in the ``deploy/cr.yaml`` configuration file is set to ``SmartUpdate``.
 
-.. note:: The manual update mode is the recomended way for a production cluster.
+When automatic updates are enabled, the Operator will carry on upgrades
+according to the following algorithm. It will query a special *Version Service* 
+server at scheduled times to obtain fresh information about version numbers and
+valid image paths needed for the upgrade. If the current version should be
+upgraded, the Operator updates the CR to reflect the new image paths and carries
+on sequential Pods deletion in a safe order, allowing StatefulSet to redeploy
+the cluster Pods with the new image.
 
-.. note:: Only the incremental update to a nearest minor version is supported
-   (for example, update from 1.2.0 to 1.3.0).
+The upgrade details are set in the ``upgradeOptions`` section of the 
+``deploy/cr.yaml`` configuration file. Make the following edits to configure
+updates:
+
+#. Set the ``apply`` option to one of the following values:
```

Comment:
We should make sure Disabled/Never is enumerated and it's clarified that when this is set automatic upgrades are disabled but the SmartUpdate functionality will continue working for other changes triggered by for instance updating a ConfigMap, rotating a password, or changing resource values.